### PR TITLE
Gradle

### PR DESCRIPTION
**Pull request by peacepassion**
#254

### CODE REVIEW

**Comment by johncarl81** [comment]
How about adding a built-in gradle wrapper?

**Comment by johncarl81** [comment]
Looks like you've moved some of the example projects, can you keep them in place?

File: parceler/src/main/java/org/parceler/ParcelAnnotationProcessor.java [hunk 1]
```diff
@@ -69,6 +69,8 @@
     public synchronized void init(ProcessingEnvironment processingEnv) {
         super.init(processingEnv);
 
+        System.out.println("################# INIT");
```
Comment (johncarl81) [inline parceler/src/main/java/org/parceler/ParcelAnnotationProcessor.java]: Remove.

**Comment by johncarl81** [comment]
A previous attempt at this: https://github.com/johncarl81/parceler/pull/102

**Comment by peacepassion** [comment]
@johncarl81 Hi, I am willing to merge this PR. I am just doing small modification for personal Android project. The feature I am developing is disabling reflection usage so I can obfuscate classed annotated by `@Parcel`.  

However, I don't know how to use maven so I have to convert this project into gradle.

**Comment by peacepassion** [comment]
But, converting maven project to gradle project is helpful. After all, gradle is more popular and more powerful.

**Comment by johncarl81** [comment]
I'm interested to hear more about your use case.  What is your end goal of obfuscating the generated classes?

**Comment by peacepassion** [comment]
@johncarl81 Currently, `Parceler` demands following proguard rule:

```
-keep interface org.parceler.Parcel
-keep @org.parceler.Parcel class * { *; }
-keep class **$$Parcelable { *; }
```

1. If the reflection can be disabled the second rule can be deleted. 
2. I will write a gradle plugin generating a map containing the relation between `**$$Parcelable` and its obfuscated result. I will inject the map into `Parcels#REPOSITORY`. So the 3rd rule can be deleted.

**Comment by johncarl81** [comment]
You can avoid reflection by not using the Parcels utility class.

**Comment by peacepassion** [comment]
But, I want to replace current warning on `private field` by error. So I can detect the wrong usage at compile stage.

**Pull request closed without merging**